sdv/datasets/demo.py

```diff
@@ -171,6 +171,9 @@ def get_available_demos(modality):
             headers = client.head_object(Bucket=BUCKET, Key=item['Key'])['Metadata']
             size_mb = headers.get('size-mb', np.nan)
             tables_info['size_MB'].append(round(float(size_mb), 2))
-            tables_info['num_tables'].append(headers.get('num-tables', np.nan))
+            num_tables = headers.get('num-tables', pd.NA)
+            if not pd.isna(num_tables):
+                num_tables = int(num_tables)
+            tables_info['num_tables'].append(num_tables)
```

Comment:
@npatki seemed fine with the 'num_tables' column being a float in the event a NaN was present which simplifies things.

I think instead of casting to in as we append, we can keep the old append approach from main, and after the 'num_tables' list is finalized, use `pd.to_numeric` to cast the whole list into a numeric array instead.  If there are `NaN` values we'll get a float array and if not we should get an int array. We can use the integration tests to confirm that the output dtype is an integer type

Reply:
Sounds reasonable to me. I'll update and add Neha to the review